File: samples/tree.js
const { create_turtle, home, clear, stop, print, state } = require('@picoturtle/picoturtle-nodejs-client');
const { font, filltext, stroketext } = require('@picoturtle/picoturtle-nodejs-client');
const { penup, pendown, penwidth, pencolour } = require('@picoturtle/picoturtle-nodejs-client');
const { left, right, forward, back } = require('@picoturtle/picoturtle-nodejs-client');
const { setpos, setx, sety, heading } = require('@picoturtle/picoturtle-nodejs-client');

let main = async () => {
    /* Create the turtle before using */
    await create_turtle();

    /* Your code goes here */

    async function tree(size) {
        if (size < 4) {
            await forward(size);
            await forward(-size);
            return;
        }

        await forward(size / 4);

        await left(60);
        await tree(size / 2);
        await right(60);

        await forward(size / 4);

        await right(35);
        await tree(size / 2);
        await left(35);

        await forward(size / 4);

        await left(30);
        await tree(size / 2);
        await right(30);

        await forward(size / 4);

        await right(25);
        await tree(size / 2);
        await left(25);

        await forward(-size);
    }

    await penup();
    await back(100);
    await pendown();
    await pencolour(165, 42, 42);
    await tree(128);

    /* Your code ends here */

    /* Always stop the turtle */
    await stop();
};

main()
.catch((err) => {
    console.error(err);
    process.exit(1);
});

File: samples/tree.cs
using System;
using picoturtle;

namespace cspico
{
    class MainClass
    {
        public static void tree(Turtle t, double size) {
            if (size < 4) {
                t.forward(size);
                t.forward(-size);
                return;
            }

            t.forward(size / 4);

            t.left(60);
            tree(t, size / 2);
            t.right(60);

            t.forward(size / 4);

            t.right(35);
            tree(t, size / 2);
            t.left(35);

            t.forward(size / 4);

            t.left(30);
            tree(t, size / 2);
            t.right(30);

            t.forward(size / 4);

            t.right(25);
            tree(t, size / 2);
            t.left(25);

            t.forward(-size);
        }

        public static void Main(string[] args)
        {
            // Create the turtle before using
            Turtle t = picoturtle.Turtle.CreateTurtle(args);
            
            if (t != null) {
                // Your code goes here

                t.penup();
                t.back(100);
                t.pendown();
                t.pencolour(165, 42, 42);
                tree(t, 128);

                // Your code ends here

                // Always stop the turtle
                t.stop();
            }
            else {
                Console.Error.WriteLine("Error: Unable to create a turtle.");
                Environment.Exit(-1);
            }
        }
    }
}
